src/context/Post.ts

```diff
@@ -199,6 +199,9 @@ export class Data {
                     details: 'Waiting for TX inclusion...',
                 })
                 await queueContext.afterTx(transaction)
+
+                let metadata: Metadata = { transaction }
```

Comment:
Always use `const` instead of `let` if you can. `let` is a sign that the variable will be changed at some point in the future. All of the `metadata` variables in this PR can be instantiated using `const`.

Also, there's not really a need for a variable here. You could `return { transaction }` or `return { transaction } as Metadata` if you really want the type